docs/badge.md

```diff
@@ -142,3 +172,52 @@ Custom component to replace the badge outer component
 |          Type          |                  Default                   |
 | :--------------------: | :----------------------------------------: |
 | React Native Component | View, if `onPress` then `TouchableOpacity` |
+
+---
+
+## Reference for withBadge
+
+### `value`
+Text value to be displayed by badge, defaults to empty
+
+|                                              Type                                       | Default |
+| :-------------------------------------------------------------------------------------: | :-----: |
+| String OR Number OR React Native Component OR Function, which returns one of the former |  none   |
+
+---
+
+### `options`
+Object with the following keys:
+
+#### `bottom`
+
+|  Type  | Default |
+| :----: | :-----: |
+| number |   10    |
+
+#### `left`
+
+|  Type  | Default |
+| :----: | :-----: |
+| number |   15    |
+
+#### `right`
+
+|  Type  | Default |
+| :----: | :-----: |
+| number |   15    |
+
+#### `top`
+
+|  Type  | Default |
+| :----: | :-----: |
+| number |   15    |
+
+#### `badgeProps` (Spreaded out)
```

Comment:
Remove this instead use similar pattern we use in other components

e.g. https://github.com/react-native-training/react-native-elements/blame/next/docs/image.md#L38 just link it to badge props instead